### ISSUE TITLE
[1.16.5] Air Compressors doesn't consume fuels

### ISSUE THREAD

**Issue by ZigTheHedge** [TeamPneumatic/pnc-repressurized#852]
Subject. Air compressors are running by its own just fine.

Minecraft: 1.16.5
Forge: 36.1.32
Mod Version: 2.13.1-202

**Comment by desht** [comment]
Er, nope, not seeing that behaviour at all.

Please provide detailed instructions on how you set your compressor up and what you did to trigger your issue.

**Comment by ZigTheHedge** [comment]
Well, I just place the Air Compressor down. Here's the short video: https://youtu.be/4gGVC9j9fMo

**Comment by desht** [comment]
That's... bizarre.  The only way this could happen is if the game thinks air is a burnable item.  What pack are you running, and are there any custom fuel definitions in a datapack?  (recipes of type `pneumaticcraft:fuel`).  It does feel a possible bad recipe somewhere.

Does the air compressor still run if you put a non-burnable item in?  Actually, does it allow non-burnable items in the fuel slot?
Does the liquid compressor exhibit similar behaviour?

I can work around this by explicitly ignoring empty itemstacks in the fuel slot, but that doesn't really explain the cause of the problem...

**Comment by ZigTheHedge** [comment]
> What pack are you running, and are there any custom fuel definitions in a datapack? (recipes of type pneumaticcraft:fuel)
It's my own pack. No, there are no any custom fuels.

> Does the air compressor still run if you put a non-burnable item in? Actually, does it allow non-burnable items in the fuel slot?
No, it's impossible to insert non-burnable item in

> Does the liquid compressor exhibit similar behaviour?
No, it works normally

I've found the mod which causes this issue - it's Advent of Ascension. Without it, the Air Compressor doesn't run with empty slot... Still have no idea how that's possible though.

**Comment by desht** [comment]
Yeah, that is weird.  Gonna have to find out how that's happening.  As I said, I can possibly work around the problem.  Pushing a dev build now - if you're in a position to test, you can try build 204 from https://jenkins.k-4u.nl/job/PneumaticCraft-Repressurized-1.16/

**Comment by desht** [comment]
A bit of investigation confirms that AoA has an event handler which returns a fuel value of 300 for empty itemstacks, so this is most definitely a bug with that mod.  However, I can also confirm that the workaround I added prevents air compressors from running on empty.

**Comment by ZigTheHedge** [comment]
Thanks! I'll open a ticket in AoA tracker about that, but thank you for the workaround!

**Comment by desht** [comment]
Keeping this open until the next release (reminds me what's been fixed)

**Comment by ZigTheHedge** [comment]
Ah, sorry :)

**Comment by Tslat** [comment]
I disagree

My event handler returns the default burn time if my custom list does not contain an entry.
This means it will always only return what the event itself is calling for.

That being said - you're not meant to be passing air to the getItemBurnTime, that should be checked on your side prior to calling the event, or by calling the event properly through ForgeHooks.getBurnTime (which does the air check for you)

**Comment by desht** [comment]
@Tslat I did some debugging of this when it was reported and I found that your `GlobalEvents#FURNACE_FUELS` had ended up with a load of "air" keys (different objects mapping to different fuel values but all displaying as "air").  This is what led to an empty ItemStack getting a burn time of 300 (via the `FurnaceFuelBurnTimeEvent`).  I couldn't see anything obviously wrong with your code, so I'm not sure how that happened.

I have since updated my code to check for an empty item, which resolves the problem for Air Compressors, at least.
